### PR TITLE
Reusable action to publish md docs to confluence

### PR DESCRIPTION
**Pull request by scheibinger**

### CODE REVIEW

File: actions/md-confluence-sync/action.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    - name: Checkout repo
+      if: inputs.checkout-repo == 'true'
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      with:
+        fetch-depth: ${{ inputs.checkout-repo-fetch-depth }}
+
+    - name: Setup tools
+      shell: bash
+      run: |
+        go install github.com/kovetskiy/mark@11.1.0
```
Comment (rafaelfelix) [inline actions/md-confluence-sync/action.yml:65]: are we sure that go is present for this to succeed?

**Review by rafaelfelix — COMMENTED** [review]
looks good, just left a comment about the present of `go` in the setup tools step

File: actions/md-confluence-sync/action.yml (hunk 1 above)
Comment (scheibinger) [reply actions/md-confluence-sync/action.yml:65]: it's included in the default toolset on github hosted runners: https://github.com/actions/runner-images/blob/main/images/ubuntu/toolsets/toolset-2404.json#L37

Not sure if I need to use setup-go action, but will see once I test it

**Review by rafaelfelix — DISMISSED** [review]

File: actions/md-confluence-sync/mark-sync.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+# $USER is your smartcontract email
+# $PASSWORD is the api token created on the https://id.atlassian.com/manage-profile/security/api-tokens page
+# Mermaid diagrams are rendered locally and inlined into the HTML artifact as PNGs
+
+mark -f "$FIILES" \
+  --edit-lock \
+  --title-from-h1 \
+  -u "$USER" \
+  -p "$TOKEN" \
+  -b https://smartcontract-it.atlassian.net/wiki/ \
```
Comment (njegosrailic) [inline actions/md-confluence-sync/mark-sync.sh:13]: Maybe this should be an input? This is a public repo, and someone else might want to use it as well.

**Review by njegosrailic — APPROVED** [review]

**Pull request merged**